.github/workflows/release.yaml

```diff
@@ -136,39 +145,40 @@ jobs:
             kustomize build "$d" --load-restrictor LoadRestrictionsNone > kustomize-$(basename "$d").yaml
           done
           kustomize build config/crd > crds.yaml
-          gh release upload "${{ github.ref_name }}" kustomize-*.yaml crds.yaml
+          gh release upload "${RELEASE_NAME}" kustomize-*.yaml crds.yaml
 
   image:
     runs-on: ubuntu-latest
     steps:
       - name: Clone repo
-        uses: actions/checkout@v4
-
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+        with:
+          persist-credentials: false
       - name: Login to GitHub Container Registry
-        uses: docker/login-action@v3
+        uses: docker/login-action@74a5d142397b4f367a81961eba4e8cd7edddf772 # v3
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Login to Quay.io
-        uses: docker/login-action@v3
+        uses: docker/login-action@74a5d142397b4f367a81961eba4e8cd7edddf772 # v3
         with:
           registry: quay.io
           username: ${{ secrets.QUAY_USERNAME }}
           password: ${{ secrets.QUAY_PASSWORD }}
 
-      - uses: actions/setup-go@v5
+      - uses: actions/setup-go@0aaccfd150d50ccaeb58ebd88d36e91967a5f35b # v5
         with:
           go-version-file: "go.mod"
-          cache: true
+          cache: false
```

Comment:
Any reason to disable cache here?

Reply:
It's possible to perform cache poisoning attacks by poisoning the cache in unprivileged workflows and then using the elevated permissions of the release workflow: https://woodruffw.github.io/zizmor/audits/#cache-poisoning